packages/mdc-menu/index.ts

```diff
@@ -22,57 +22,51 @@
  */
 
 import MDCComponent from '@material/base/component';
+import {CustomEventListener, SpecificEventListener} from '@material/base/types';
+import {ListActionEvent, MDCList, MDCListFoundation} from '@material/list/index';
+import {MDCMenuSurfaceFoundation} from '@material/menu-surface/foundation';
+import {Corner, MDCMenuSurface} from '@material/menu-surface/index';
+import {MenuDistance} from '@material/menu-surface/types';
+import {cssClasses, strings} from './constants';
 import {MDCMenuFoundation} from './foundation';
-import {strings, cssClasses} from './constants';
-import {MDCMenuSurface, Corner} from '@material/menu-surface/index.ts';
-import {MDCMenuSurfaceFoundation, MenuDistance} from '@material/menu-surface/foundation.ts';
-import {MDCList, MDCListFoundation} from '@material/list/index';
+import {ListFactory, MenuItemComponentEventDetail, MenuSurfaceFactory} from './types';
 
-/**
- * @extends MDCComponent<!MDCMenuFoundation>
- */
-class MDCMenu extends MDCComponent {
-  /** @param {...?} args */
-  constructor(...args) {
-    super(...args);
-    /** @private {!MDCMenuSurface} */
-    this.menuSurface_;
-    /** @private {!MDCList} */
-    this.list_;
-    /** @private {!Function} */
-    this.handleKeydown_;
+class MDCMenu extends MDCComponent<MDCMenuFoundation> {
+  static attachTo(root: Element) {
+    return new MDCMenu(root);
+  }
 
-    /** @private {!Function} */
-    this.handleItemAction_;
+  private menuSurfaceFactory_!: MenuSurfaceFactory; // assigned in initialize()
+  private listFactory_!: ListFactory; // assigned in initialize()
 
-    /** @private {!Function} */
-    this.afterOpenedCallback_;
-  }
+  private menuSurface_!: MDCMenuSurface; // assigned in initialize()
```

Comment:
I don't think the `//` comments are necessary. 

But the `menuSurface_` and `list_` are assigned in `initialSyncWithDom`

Reply:
The comments are a good idea for anything using `!` to explain why it's there.

Reply:
Done